File: src/main/java/klaxon/akka/examples/java/future/FibonacciActor.java
package klaxon.akka.examples.java.future;

import akka.actor.UntypedActor;

/**
 * <p>date 9/17/14 </p>
 *
 * @author klaxon
 */
public class FibonacciActor extends UntypedActor {

    @Override
    public void onReceive(Object message) throws Exception {
        if (message instanceof Integer) {
            int calculated = calculateFibonacci((Integer) message);
            sender().tell(calculated, self());
        }
    }

    private int calculateFibonacci(int number) {
        if (number <= 0) return 0;
        if (number == 1) return 1;
        return calculateFibonacci(number - 1) + calculateFibonacci(number - 2);
    }

}

File: src/main/scala/klaxon/akka/examples/scala/FibonacciActor.scala
package klaxon.akka.examples.scala

import java.util.concurrent.TimeUnit

import akka.actor.Actor

/**
 * <p>date 9/17/14 </p>
 * @author klaxon
 */
class FibonacciActor extends Actor {

  val fibs: Stream[Int] = 0 #:: fibs.scanLeft(1)(_ + _)

  override def receive: Receive = {
    case number: Int =>
      TimeUnit.SECONDS.sleep(1)
      sender() ! fibs(number)
  }

}